syntax = "proto3";

package example.v1;

option go_package = "github.com/productscience/chain-protos/network_node/v1";

service RecordService {
  rpc SetRecord(SetRecordRequest) returns (SetRecordResponse);

  rpc GetRecord(GetRecordRequest) returns (GetRecordResponse);
}

message Record {
  string key = 1;
  string value = 2;
}

message SetRecordRequest {
  Record record = 1;
}

message SetRecordResponse {
  bool success = 1;
}

message GetRecordRequest {
  string key = 1;
}

message GetRecordResponse {
  Record record = 1;
}
